### DISCUSSION TITLE
画像を指で移動させる際のエラーについて

### DISCUSSION THREAD

**Discussion by ShunHigu** [lit-iphone-course/lit-iphone-course#54] [Question-original]
１　今、写真をボタンで増加させ、それを動かすコードを書いています。StoneImage[1]のように配列を使うことで画像を識別できるようにしているのですが、なぜかエラーが出ていないにも関わらずページ移動ができません。

２　画像を消え巣に表示させることはできたのですが、その後がうまくいきませんでした。

３　移動はこちらを参考にしました
https://i-app-tec.com/ios/image-drag.html

４　 @IBAction func Renda(){
        num=Int.random(in: 1...100)
        let xrand=Double.random(in: 40...370)
        let yrand=Double.random(in: 40...450)
       
        if num<=75{
        	let image1:UIImage = UIImage(named:"Stone")!
            StoneImage[stonenum]=UIImageView(image:image1)
            StoneImage[stonenum].center = CGPoint(x:xrand, y:yrand)
            view.addSubview(StoneImage[stonenum])
            stonenum=stonenum+1
        }
        if (num>=76)&&(num<=95){
            let image1:UIImage = UIImage(named:"Stone")!
            EnemyImage[enemynum]=UIImageView(image:image1)
            EnemyImage[enemynum].center = CGPoint(x:xrand, y:yrand)
            view.addSubview(EnemyImage[enemynum])
            enemynum=enemynum+1
        }
    }
    //　ドラッグ時に呼ばれる
    override func touchesMoved(_ touches: Set<UITouch>, with event: UIEvent?){
        for i in 0...stonenum{
            // タッチイベントを取得
            let touchEvent = touches.first!
            // ドラッグ前の座標, Swift 1.2 から
            let preStonex = touchEvent.previousLocation(in: self.view).x
            let preStoney = touchEvent.previousLocation(in: self.view).y
            // ドラッグ後の座標
            let newStonex = touchEvent.location(in: self.view).x
            let newStoney = touchEvent.location(in: self.view).y
            // ドラッグしたx座標の移動距離
            let Stonedx = newStonex - preStonex
            // ドラッグしたy座標の移動距離
            let Stonedy = newStoney - preStoney
            // 画像のフレーム
            var viewFrame: CGRect = StoneImage[i].frame
            // 移動分を反映させる
            viewFrame.origin.x += Stonedx
            viewFrame.origin.y += Stonedy
            StoneImage[i].frame = viewFrame
            self.view.addSubview(StoneImage[i])
        }
        
        
    }
    
    @IBOutlet var HPText:UILabel!
    
    var HP:Int=100
 
    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(true)
        timer.invalidate()
    }
    
    var i:Int!
    @objc func update(tm: Timer) {
        for i in 0...enemynum{
            if EnemyImage[i].isHidden==false{
                //繰り返し動作
                HP-=10
                HPText.text=("\(HP)/100")
            }
        }
    }
    
５　https://github.com/ShunHigu/Original
↑新しく更新されてますか？

**Comment by Yonpachi48** [comment]
initial commitから変わっていないのでうまくcommitができてなさそう？
それと、確認なんだけどXcodeのバージョンを教えていただきたいです、！

**Reply by ShunHigu** [reply]
initial commitとは何のことでしょうか？調べてみても良い記事がなくてどこからinitial commitを変えれるでしょうか？
verion14.0です!

**Reply by Yonpachi48** [reply]
GitHub入門の教科書のCommit&Pushのところを参考にして一度やってみて〜！

**Reply by ShunHigu** [reply]
できました！お願いします！
https://github.com/ShunHigu/Original
